packages/analytics-js-common/__tests__/utilities/page.test.ts

```diff
@@ -182,4 +182,21 @@ describe('onPageLeave', () => {
 
     expect(evCallback).toHaveBeenNthCalledWith(2, false);
   });
+
+  it('should fire the callback on beforeunload event on the next tick even when the tab is inactive', () => {
+    const evCallback = jest.fn();
+    onPageLeave(evCallback);
+
+    setVisibilityState('hidden');
+    dispatchDocumentEvent('visibilitychange');
+
+    dispatchWindowEvent('beforeunload');
+
+    expect(evCallback).toHaveBeenCalledTimes(1);
+    expect(evCallback).toHaveBeenNthCalledWith(1, true);
+    setTimeout(() => {
+      expect(evCallback).toHaveBeenCalledTimes(1);
+      expect(evCallback).toHaveBeenNthCalledWith(1, false);
```

Comment:
I think you meant to check if the callback was called again. But it doesn't work like that.
The mock accumulates the number of times it was called until it is reset.

You have two options:
- Reset the mock so that the invocation counters are cleared
- Check that it is called 2 times and verify the values for second invocation.

Reply:
Addressed